frontend/src/lib/components/LemonSelect.tsx

```diff
@@ -135,15 +138,32 @@ export function LemonSelect<T>({
 
     const [sections, allLeafOptions] = useMemo(() => boxToSections(options), [options])
 
+    const elementsRef = useRef(allLeafOptions.map(() => createRef<HTMLButtonElement>()))
+
+    useEffect(() => {
+        if (focusedOptionIndex > -1) {
+            elementsRef.current?.[focusedOptionIndex]?.current?.focus()
+        }
+    }, [focusedOptionIndex])
+
+    const handleKeyboardEvents = (e: React.KeyboardEvent<HTMLElement>): void => {
+        if (e.key === 'ArrowDown' && focusedOptionIndex < allLeafOptions.length - 1) {
+            setFocusedOptionIndex(focusedOptionIndex + 1)
+        } else if (e.key === 'ArrowUp' && focusedOptionIndex > 0) {
+            setFocusedOptionIndex(focusedOptionIndex - 1)
+        }
+    }
+
     return (
-        <div className="flex">
+        <div className="flex" tabIndex={-1} ref={selectRef}>
             <LemonButtonWithPopup
                 className={clsx(className, isClearButtonShown && 'LemonSelect--clearable')}
+                onKeyDown={(e) => handleKeyboardEvents(e)}
                 popup={{
                     ref: popup?.ref,
                     overlay: sections.map((section, i) => (
-                        <>
-                            <div key={i} className="space-y-px">
+                        <div key={i}>
```

Comment:
Can't you just add the key to the div below? 🤔

Reply:
It was complaining 🤷‍♀️

Reply:
I think that's because before there was a React.Fragment outside the div.

Perhaps shockingly, the only attribute a fragment does accept is the key 😅 

https://reactjs.org/docs/fragments.html#keyed-fragments

Reply:
I did not know that! Learn something new every day 🧠